test/e2e/operator_test.go

```diff
@@ -418,20 +452,40 @@ func TestContourSpec(t *testing.T) {
 	}
 	t.Logf("observed expected status conditions for contour %s/%s", operatorNs, testName)
 
-	if err := newClusterIPService(ctx, kclient, appName, specNs, 80, 8080); err != nil {
-		t.Fatalf("failed to create service %s/%s: %v", specNs, appName, err)
+	if err := newClusterIPService(ctx, kclient, appName, cfg.SpecNs, 80, 8080); err != nil {
+		t.Fatalf("failed to create service %s/%s: %v", cfg.SpecNs, appName, err)
 	}
-	t.Logf("created service %s/%s", specNs, appName)
+	t.Logf("created service %s/%s", cfg.SpecNs, appName)
 
-	if err := newIngress(ctx, kclient, appName, specNs, appName, 80); err != nil {
-		t.Fatalf("failed to create ingress %s/%s: %v", specNs, appName, err)
+	if err := newIngress(ctx, kclient, appName, cfg.SpecNs, appName, 80); err != nil {
+		t.Fatalf("failed to create ingress %s/%s: %v", cfg.SpecNs, appName, err)
 	}
-	t.Logf("created ingress %s/%s", specNs, appName)
+	t.Logf("created ingress %s/%s", cfg.SpecNs, appName)
 
-	if err := waitForHTTPResponse(testURL, 1*time.Minute); err != nil {
-		t.Fatalf("failed to receive http response for %q: %v", testURL, err)
+	// testURL is the url used to test e2e functionality.
+	testURL := "http://local.projectcontour.io/"
```

Comment:
It looks like a lot of these tests are the same or very similar? Could we dry these up a bit, either here or in a followup?

Reply:
+1